BlueprintUI/Tests/XCTestCaseAdditions.swift

```diff
@@ -0,0 +1,21 @@
+//
+//  XCTestCaseAdditions.swift
+//
+//
+//  Created by Kyle Van Essen on 11/3/22.
+//
+
+import Foundation
+import XCTest
+
+extension XCTestCase {
+
+    /// Provides scoping within a single `func test...()` method, eg
+    /// if you want to test multiple cases / permutations with a label.
+    public func testcase(
```

Comment:
Why not just use `do {` with a comment?

Reply:
I have this in Market and Listable. It's a nicer way to define sub-cases vs the comment and IMO somewhat abuse of the `do {`.

Reply:
IMO it's weird that the `String` is unused—it looks like it should do more than just create a scope.

I don't feel strongly, but I try to avoid stuff like this since it interrupts code review while you spin up on what this does, where just using the native patterns/tools is instantly obvious.